packages/typespec-ts/src/transform/transformSchemas.ts

```diff
@@ -112,7 +112,7 @@ export function transformSchemas(client: SdkClient, dpgContext: SdkContext) {
     }
   }
   transformHostParameters();
-  program.stateMap(modelKey).forEach((context, tspModel) => {
+  usageMap.forEach((context, tspModel) => {
```

Comment:
Confirmed with Timothee, this is abuse of program.stateMap which is only used for saving decorator info: https://github.com/microsoft/typespec/issues/3335#issuecomment-2107933123 

So I replaced it with internal map instead: .